activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -98,6 +99,9 @@ def new_client(conn_params)
       #   ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.create_unlogged_tables = true
       class_attribute :create_unlogged_tables, default: false
 
+      cattr_accessor :additional_type_records_cache, default: []
```

Comment:
Instead of having those variables here we should change the implementation that need those values to always ask the value to the schema cache and the schema cache delegate to the adapter if the value is not present. In the same way we do with table information.

https://github.com/rails/rails/blob/51cc5ccd03428bab9863d5e8ae514661c2bd2b57/activerecord/lib/active_record/model_schema.rb#L380
https://github.com/rails/rails/blob/c9c293e19464c9ce37316f2d917ac123d7b654e6/activerecord/lib/active_record/connection_adapters/schema_cache.rb#L84